### PR TITLE
perf(models): fix census hot paths and bounded fixture drift

### PR DESCRIPTION
**Pull request by ooples**
## Goal  Fix every actionable model-performance census finding without changing the production architectures or defaults defined by the original research. CI may bound valid synthetic input geometry or repetition count, but it may not truncate or replace the model being tested.  ## In progress / ownership  This PR is owned by `agent/perf-census-hotpaths`; please do not duplicate these items in another branch.  - [x] generated-fixture geometry and non-expansion contracts - [x] MedCLIP, SVTR, UPR-Net, and Upscale-A-Video paper contracts - [x] exact architecture, serialization, clone, and parameter-enumeration contracts - [x] all original census errors and warnings investigated with isolated-process measurements - [x] PerfView/EventPipe hot-path study - [x] shared Mamba depthwise-convolution engine path - [x] frozen-backbone activation checkpointing with selected temporal gradients - [x] Tensors #951 six-backend UPR-Net operations - [x] Tensors #952 Conv3D, checkpoint, gradient-pruning, compiler, and GEMM fixes - [x] complete 863-record integration census: zero errors and zero warnings - [x] merge/release Tensors #951/#952 and consume the public 0.127.0 package set - [x] package-only local build, focused integration suite, and review-thread audit - [ ] fresh package-only GitHub Actions at commit `d2437d930`  ## Root causes and fixes

- The census child loaded Tensors before the in-process CPU opt-out, compiling hundreds of OpenCL kernels and inflating small-model memory. The runner now sets `AIDOTNET_DISABLE_GPU=1` before process start.
- MetaCLIP's generated image tensor disagreed with its bounded public options, quadrupling the token count and making attention maps 16x larger. Its fixture now honors the configured 32x32 contract.
- Silero VAD received four streaming frames as one sample. Its fixture now uses one valid 512-sample 16 kHz frame.
- Mamba expanded every causal depthwise convolution into `sequence * kernel` tape nodes. It now uses one differentiable `IEngine.DepthwiseConv1D` operation with exact causal padding and unchanged weights.
- StableVideoSR retained the full frozen spatial backbone and computed gradients that could never be returned. Each paper-defined block is checkpointed; recomputation requests only the 105 inserted temporal adapters while preserving the input VJP through every frozen spatial operation.
- StableVideoSR now registers its native diffusion core under a stable component ID, so checkpointing, optimizers, cloning, and the census all see the same 733,249,679-parameter surface instead of reporting zero.
- The generated StableVideoSR constructor and fixture shape are wired in lockstep; the minimum valid 8x8 geometry traverses every four-stage spatial level and the four-frame temporal path without changing the released graph.
- The census counts readiness-aware manifest slots without materializing the 733M-parameter vector or repeatedly walking it for iteration projections.
- Tensors 0.127.0 removes the remaining Conv3D, selective-linear, checkpoint-arena, lazy-compiler, and 32-core GEMM-routing hot paths.
- AiDotNet's Conv3D layer routes training and compiled traces through the package-native differentiable `IEngine.Conv3D` operation; the duplicate consumer-side im2col/col2im graph implementation was deleted.
- The Upscale-A-Video fixture exercises the real conditioned inference/training contract, including activation checkpointing and a verified temporal-parameter update, rather than sending RGB conditions through the generic latent `Predict` contract.
- Baselines include CPU model identity and a schema version, preventing incomparable hardware records from being treated as regressions while still enforcing every absolute ceiling.
- Cohort analysis detects abnormal train/steady-forward amplification; absolute train, forward, wall, and memory ceilings remain unchanged.
## Paper fidelity

The final StableVideoSR fixture still uses:

- exact `[256,512,512,1024]` four-stage video U-Net
- 22 spatial residual blocks, 16 Transformer3D blocks, 10 only-cross-attention blocks, 9 temporal modules, 3 downsamplers, and 3 upsamplers
- exact Stable Diffusion x4 CLIP encoder: width 1024, 23 layers, 16 heads, projection width 512
- 733,249,679 parameters
- four frames, bounded `8x8` low-resolution census input, and `32x32` output (production geometry/defaults unchanged)
- high-resolution temporal-VAE latent as the clean target
- separately noised three-channel low-resolution RGB condition and seven-channel U-Net input
- distinct DDPM timestep per batch element
- frozen VAE, CLIP, and spatial U-Net; only the paper's temporal adapters are optimized

The PR also preserves the released MedCLIP, SVTR, and UPR-Net defaults documented in the model contract tests. No production width, depth, vocabulary, stage, or parameter count was reduced for performance.
## Complete census evidence  The original isolated-process run produced **1 timeout and 9 warnings**. The final merged ledger contains **863/863 durable records, 863 ok, 0 errors, and 0 warnings**.  | Original finding | Before | Final | |---|---:|---:| | StableVideoSR | timeout; 24,185.9 MiB peak | 42,606.5 ms train; 19,847.9 MiB peak | | SegMamba | 9,748.1 ms train; 606.5 MiB allocated | 594.8 ms; 155.2 MiB | | MGLDVSR | 7,292.9 ms train | 6,453.3 ms | | SileroVad | 659.4 ms train; 247.7 MiB peak | 401.2 ms; 107.7 MiB | | MetaCLIP | 525.5 ms train; 389.9 MiB allocated; 466.2 MiB peak | 320.7 ms; 43.0 MiB; 88.9 MiB | | GraphCodeBERT | 186.9 MiB peak | 85.5 MiB | | GraFPrint | 182.1 MiB peak | 92.8 MiB | | APNet2 | 260.6 MiB peak | 150.4 MiB | | SpikingNeuralNetwork | 245.0 MiB peak | 139.4 MiB |  ### Exact StableVideoSR proof

| Measurement | Original | Final 0.127 wiring |
|---|---:|---:|
| result | timeout | ok; production aggregate: 0 errors, 0 warnings |
| runner elapsed | 214.3 s | 104.02 s |
| measured wall | 208.91 s | 102.02 s |
| cold forward | 57+ s | 17.69 s |
| steady forward p95 | 30+ s | 29.51 s |
| train step | 56.97 s | 41.67 s |
| allocated bytes | 127.12 GB | 101.65 GB total; bounded 11.14 GiB peak working set |
| peak private memory | 57.9 GiB | 11.46 GiB |
| parameter surface | previously hidden by outer wrapper | 733,249,679 across 44 stable slots |
| execution backend | contaminated by OpenCL discovery | CPU-only before assembly load; no OpenCL compilation |

Its final training step is only **1.41x** its measured steady forward, so the training path is healthy relative to the work the exact model performs.
## Validation

- public NuGet restore resolved `AiDotNet.Tensors`, `AiDotNet.Native.OpenBLAS`, `AiDotNet.Native.OneDNN`, and `AiDotNet.Native.CLBlast` at `0.127.0`; no project reference or local package source was used
- AiDotNet Release/net8.0 package-only build: 0 errors
- AiDotNet Release/net10.0 package-only build: 0 errors
- AiDotNet.Tests Release/net10.0 package-only build: 0 errors
- focused 0.127.0 consumer suite: 24 passed, 0 failed, 0 skipped
  - package-native Conv3D forward parity plus finite/nonzero input, kernel, and bias gradients
  - selective checkpoint source factory executes after forward materialization; input VJP and selected parameter gradient are retained while frozen parameters are omitted
  - all 13 Upscale-A-Video family tests, including real conditioned training with activation checkpointing and a changed temporal-parameter fingerprint
  - all 5 UPR-Net paper/integration contracts
  - VoxelCNN fused compiled-training update/no-NaN regression in 2 seconds
- post-wiring focused net10 suite: 16 passed, 0 failed (Conv3D, Upscale-A-Video, MedCLIP, and selective checkpoint gradients)
- Release/net471 portability build: 0 errors
- ModelPerfProbe self-test: passed
- exact StableVideoSR production aggregate: 1/1 fixture, 0 errors, 0 warnings
- complete integration census aggregation: 863/863, 0 errors, 0 warnings
- Tensors Release/net10.0 build: 0 warnings, 0 errors
- Tensors full autodiff suite: 909 passed, 29 skipped, 0 failed
- new Tensors compiler-scaling and GEMM-routing checks: 11 passed, 0 failed
- review-thread audit at `d2437d930`: 71 total, 0 unresolved threads
## Tensors 0.127.0 integration

The dependency chain is complete: Tensors #951/#952 and release PR #954 are merged, the `v0.127.0` release pipeline succeeded, and all four lockstep packages restore from the public NuGet feed. This PR consumes those packages directly and wires package-native Conv3D, selective checkpointing, stable parameter manifests, and CPU-only census startup into real AiDotNet execution.

No temporary or absolute Tensors project reference is committed. The only remaining gate is the fresh package-only GitHub Actions run for `d2437d930`.

## New Features
- Added native Upscale-A-Video and Stable Video SR support with prompt conditioning, temporal windows, guidance, and optical-flow propagation.
- Added video diffusion transformers, temporal modules, attention, and trainable projection layers.
- Added configurable SVTR-Tiny OCR with optional TPS rectification.
- Added MedCLIP semantic matching losses, configurable tokenization, and learned temperature scaling.
- Added UPR-Net and video U-Net architecture profiles.
- Added configurable UNITER training optimization settings and new learnable embedding layers.

## Bug Fixes
- Improved shape validation, model training, serialization, cloning, parameter restoration, and causal sequence processing.
- Improved checkpoint compatibility and parameter matching across model changes.

### CODE REVIEW

File: src/NeuralNetworks/NeuralNetworkBase.cs [hunk 1]
```diff
@@ -14004,6 +14004,18 @@ public virtual void ConfigureFairness(Vector<int> sensitiveFeatures, params Fair
...
+    /// </summary>
+    /// <remarks>
+    /// This is deliberately an explicit opt-in. A serialization-order layer list does not prove a
+    /// sequential graph: combined-stream, encoder/decoder, shared-stage, and input-transforming
+    /// models can all inherit the base lazy-shape resolver while owning non-linear execution
+    /// boundaries. The safe default records the model's actual forward. A plain-chain base may
+    /// override this only when forwarding <c>Layers[0..n]</c> exactly reproduces <see cref="Predict"/>.
+    /// </remarks>
+    protected virtual bool SupportsSequentialActivationFold => false;
+
```
[the remark this answers is not in the document]
Comment (ooples) [reply src/NeuralNetworks/NeuralNetworkBase.cs:14018]: Implemented and verified in c928b7f61. Added SequentialVectorModelLayoutBase as the reusable execution-topology contract, and moved the two audited literal-chain models (NeuralNetwork and FeedForwardNeuralNetwork) onto it. The base enables the fold only for a model directly choosing that contract; customized derived types conservatively fall back to observing their actual forward. Branched, encoder/decoder, shared-stage, input-transforming, and generation-loop models remain on the default observer path. Added async tests proving both the fast path and derived-model fallback; the combined activation/input-contract suite passes 37/37, and net471/net10.0 builds pass.
